source/server/listener_impl.cc

```diff
@@ -372,6 +375,21 @@ void ListenerImpl::buildUdpListenerFactory(Network::Socket::Type socket_type,
   }
 }
 
+void ListenerImpl::buildUdpWriterFactory(Network::Socket::Type socket_type) {
+  if (socket_type == Network::Socket::Type::Datagram) {
+    auto udp_writer_config = config_.udp_writer_config();
+    if (!Api::OsSysCallsSingleton::get().supportsUdpGso() || udp_writer_config.name().empty()) {
+      udp_writer_config.set_name(std::string(Network::DefaultWriterName));
```

Comment:
Since we are now using typed_config you actually don't want to be using name here. You should set the type_url in the defaulting case. So I don't think you need the constant name, and can just set this to the specific type_url you want?

Reply:
Replaced name with type_url here!